libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/voice/NavigationSpeechListener.java

```diff
@@ -0,0 +1,31 @@
+package com.mapbox.services.android.navigation.ui.v5.voice;
+
+import timber.log.Timber;
+
+class NavigationSpeechListener implements SpeechListener {
+
+  private SpeechPlayerProvider speechPlayerProvider;
+  private SpeechAudioFocusManager audioFocusManager;
+
+  NavigationSpeechListener(SpeechPlayerProvider speechPlayerProvider,
+                           SpeechAudioFocusManager audioFocusManager) {
+    this.speechPlayerProvider = speechPlayerProvider;
+    this.audioFocusManager = audioFocusManager;
+  }
+
+  @Override
+  public void onStart() {
+    audioFocusManager.requestAudioFocus();
+  }
+
+  @Override
+  public void onDone() {
+    audioFocusManager.abandonAudioFocus();
+  }
+
+  @Override
+  public void onError(String errorText, SpeechAnnouncement speechAnnouncement) {
+    Timber.e(errorText);
+    speechPlayerProvider.retrieveAndroidSpeechPlayer().play(speechAnnouncement);
```

Comment:
Isn't this going to go into an infinite loop if `AndroidSpeechPlayer` returns an error?

Reply:
Currently not firing error listener for `AndroidSpeechPlayer` https://github.com/mapbox/mapbox-navigation-android/blob/master/libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/voice/UtteranceListener.java#L31

Reply:
It looks like we are logging this currently in `master` https://github.com/mapbox/mapbox-navigation-android/blob/master/libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/voice/AndroidSpeechPlayer.java#L44 

Does sticking with that work okay?